### ISSUE TITLE
Dependent packages for both BFF and Identity servers have different versions.

### ISSUE THREAD

**Issue by LilitHakobyan** [DuendeSoftware/Support#1382]
**Which version of Duende BFF are you using?**
BFF version 2.2.0 IdentityServer version 7.0.6
**Which version of .NET are you using?** .NET 8

Hello,
Why do the dependent package "Microsoft.AspNetCore.Authentication.OpenIdConnect" and sub dependency "System.IdentityServer.Tokens.Jwt" for both BFF and Identity servers have different versions?
For Bff version is 7.0.3 and for Identity server version is 7.1.2.

![image](https://github.com/user-attachments/assets/c1f5accd-f1c8-4e9b-8b38-6204bb0a8e8b)
![image](https://github.com/user-attachments/assets/db95f5c8-2311-47db-857d-52ba11eaee87)

May this cause the production build issues?

I am getting deployment errors regarding the 'Microsoft.IdentityModel.Tokens" package 
For one project where BFF installed  is 
  ```
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.IdentityModel.Tokens, Version=7.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

      File name: 'Microsoft.IdentityModel.Tokens, Version=7.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
         at Duende.LicenseValidator.ValidateKey(String licenseKey)
         at Duende.LicenseValidator.Initalize(ILoggerFactory loggerFactory, String productName, String key, Boolean isDevelopment) in /_/src/Duende.Bff/Licensing/LicenseValidator.cs:line 39
```

and for another where identity server package is installed is 
```

System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.IdentityModel.Tokens, Version=7.1.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

      File name: 'Microsoft.IdentityModel.Tokens, Version=7.1.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
         at Duende.LicenseValidator`1.ValidateKey(String licenseKey)
         at Duende.LicenseValidator`1.Initialize(ILoggerFactory loggerFactory, String productName, String key) in /_/src/IdentityServer/Licensing/LicenseValidator.cs:line 55
```

Is Duende licensing involved here? We have a valid business license for Duende software.

Thanks

**Comment by RolandGuijt** [comment]
This doesn't seem related to licensing.

Within one project it's important to use the same versions of Microsoft.IdentityModel.* as we use in our packages. We get a lot of issues from customers where a mismatch leads to problems. Here's a [doc](https://docs.duendesoftware.com/identityserver/v7/troubleshooting/wilson) that describes this. The potential problems don't include the error you're describing however.

Having two different Microsoft.IdentityModel.* versions in two separate projects shouldn't be an issue. You did mention these are deployment errors. Does this work as normal on a dev machine? And where are you trying to deploy to?

**Comment by LilitHakobyan** [comment]
Yes, it works as expected in the local dev environment. 
We are trying to deploy in kubernetes.

**Comment by LilitHakobyan** [comment]
With an updated Identity server and BFF,  when running the command  "dotnet list package --include-transitive | sls "Microsoft.IdentityModel|System.IdentityModel"" I got two different versions 6.35.0 and 7.1.2 , changed some package versions and now it's all in  7.1.2 

`   > Microsoft.IdentityModel.Protocols.OpenIdConnect      7.1.2       7.1.2
   > Microsoft.IdentityModel.Abstractions                       7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                      7.1.2
   > Microsoft.IdentityModel.Logging                            7.1.2
   > Microsoft.IdentityModel.Protocols                          7.1.2
   > Microsoft.IdentityModel.Tokens                             7.1.2
   > System.IdentityModel.Tokens.Jwt                            7.1.2
   > Microsoft.IdentityModel.Abstractions                       7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                      7.1.2
   > Microsoft.IdentityModel.Logging                            7.1.2
   > Microsoft.IdentityModel.Protocols                          7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect            7.1.2
   > Microsoft.IdentityModel.Tokens                             7.1.2
   > System.IdentityModel.Tokens.Jwt                            7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                 7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                7.1.2
   > Microsoft.IdentityModel.Logging                      7.1.2
   > Microsoft.IdentityModel.Protocols                    7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect      7.1.2
   > Microsoft.IdentityModel.Tokens                       7.1.2
   > System.IdentityModel.Tokens.Jwt                      7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                       7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                      7.1.2
   > Microsoft.IdentityModel.Logging                            7.1.2
   > Microsoft.IdentityModel.Protocols                          7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect            7.1.2
   > Microsoft.IdentityModel.Tokens                             7.1.2
   > System.IdentityModel.Tokens.Jwt                            7.1.2
   > Microsoft.IdentityModel.Abstractions                   7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                  7.1.2
   > Microsoft.IdentityModel.Logging                        7.1.2
   > Microsoft.IdentityModel.Protocols                      7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect        7.1.2
   > Microsoft.IdentityModel.Tokens                         7.1.2
   > System.IdentityModel.Tokens.Jwt                        7.1.2
   > Microsoft.IdentityModel.Abstractions                     7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                    7.1.2
   > Microsoft.IdentityModel.Logging                          7.1.2
   > Microsoft.IdentityModel.Protocols                        7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect          7.1.2
   > Microsoft.IdentityModel.Tokens                           7.1.2
   > System.IdentityModel.Tokens.Jwt                          7.1.2
   > Microsoft.IdentityModel.Abstractions                 7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                7.1.2
   > Microsoft.IdentityModel.Logging                      7.1.2
   > Microsoft.IdentityModel.Protocols                    7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect      7.1.2
   > Microsoft.IdentityModel.Tokens                       7.1.2
   > System.IdentityModel.Tokens.Jwt                      7.1.2
   > Microsoft.IdentityModel.Abstractions                     7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                    7.1.2
   > Microsoft.IdentityModel.Logging                          7.1.2
   > Microsoft.IdentityModel.Protocols                        7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect          7.1.2
   > Microsoft.IdentityModel.Tokens                           7.1.2
   > System.IdentityModel.Tokens.Jwt                          7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                         7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                        7.1.2
   > Microsoft.IdentityModel.Logging                              7.1.2
   > Microsoft.IdentityModel.Protocols                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect              7.1.2
   > Microsoft.IdentityModel.Tokens                               7.1.2
   > System.IdentityModel.Tokens.Jwt                              7.1.2
   > Microsoft.IdentityModel.Abstractions                       7.1.2
   > Microsoft.IdentityModel.JsonWebTokens                      7.1.2
   > Microsoft.IdentityModel.Logging                            7.1.2
   > Microsoft.IdentityModel.Protocols                          7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect            7.1.2
   > Microsoft.IdentityModel.Tokens                             7.1.2
   > System.IdentityModel.Tokens.Jwt                            7.1.2
   > Microsoft.IdentityModel.Protocols.OpenIdConnect      7.1.2       7.1.2
   > Microsoft.IdentityModel.Abstractions            7.1.2
   > Microsoft.IdentityModel.JsonWebTokens           7.1.2
   > Microsoft.IdentityModel.Logging                 7.1.2
   > Microsoft.IdentityModel.Protocols               7.1.2
   > Microsoft.IdentityModel.Tokens                  7.1.2
   > System.IdentityModel.Tokens.Jwt                 7.1.2`

but the deployment issues still exist.
![image](https://github.com/user-attachments/assets/b0e6b2ca-312b-4a33-ba58-10c4fc2e4091)

![image](https://github.com/user-attachments/assets/ce693678-1cdf-4a2b-8a6d-7f69ec2d0a98)

Do you have any idea how to fix this issue?

**Comment by RolandGuijt** [comment]
Can you please check if the assemblies that cannot be loaded are on the actual pod and if it has the rights to read it?

**Comment by LilitHakobyan** [comment]
I am having difficulties checking the dlls on pos as the pods have CrashLoopBackOff status. 
Is there anything else that I can check meanwhile?

**Comment by RolandGuijt** [comment]
Nothing I can think of atm. Can you please report back as soon as you can do the checks?

**Comment by LilitHakobyan** [comment]
Yes, will let you know under this request.

**Comment by LilitHakobyan** [comment]
So when added all dependant packages that we saw on pod errors the deployed services started running correctly. 
Question: Is this something connected to the identity server and BFF upgrades?

**Comment by RolandGuijt** [comment]
We don't currently have reports that these packages have problems with dependencies in combination with Kubernetes. What does your deployment pipeline look like? Could it be something related to that?

**Comment by RolandGuijt** [comment]
@LilitHakobyan Were you able to solve this? If so I would like to close the issue.

**Comment by LilitHakobyan** [comment]
Yes, I was able to solve this.  
So the docker files for each project using this location \Release\net8.0\publish\win-x64 to copy the DLLs to the deployment server.
So the errors in the pod’s were related to the Transitive Dependencies and those dependent packages weren’t being updated in the location above.
To fix the local environment we deleted all files from those locations before deployment.
I also did NuGet cache clean up  dotnet nuget locals all --clear .
